server/src/main/java/org/elasticsearch/cluster/InternalClusterInfoService.java

```diff
@@ -473,6 +469,32 @@ private void callListeners() {
                 onRefreshComplete(this);
             }
         }
+
+        private ClusterInfo updateAndGetCurrentClusterInfo() {
+            final IndicesStatsSummary indicesStatsSummary = this.indicesStatsSummary; // single volatile read
+            final Map<String, EstimatedHeapUsage> estimatedHeapUsages = new HashMap<>();
+            final var currentMaxHeapPerNode = this.maxHeapPerNode; // Make sure we use a consistent view
```

Comment:
do we still need to take a copy? I thought by making these local to the async refresh we'd isolate ourselves from other refreshes? I might have missed something

Reply:
yes sorry, poor use of words, I meant reference :)